operator/docs/enhancements/ruler_support.md

```diff
@@ -0,0 +1,686 @@
+    // +kubebuilder:default:=200
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Maximum Shards"
+    MaxShards int32 `json:"maxShards"`
+
+    // Minimum number of shards, i.e. amount of concurrency.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:=200
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Minimum Shards"
+    MinShards int32 `json:"minShards"`
+
+    // Maximum number of samples per send.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:=500
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Maximum Shards per Send"
+    MaxSamplesPerSend int32 `json:"maxSamplesPerSend"`
+
+    // Maximum time a sample will wait in buffer.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:="5s"
+    BatchSendDeadline int32 `json:"batchSendDeadline"`
+
+    // Initial retry delay. Gets doubled for every retry.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:="30ms"
+    MinBackOffPeriod int32 `json:"minBackOffPeriod"`
+
+    // Maximum retry delay.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:="100ms"
+    MaxBackOffPeriod int32 `json:"maxBackOffPeriod"`
+}
+
+// RemoteWriteSpec defines the configuration for ruler's remote_write connectivity.
+type RemoteWriteSpec struct {
+    // Enable remote-write functionality.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch",displayName="Enabled"
+    Enabled bool `json:"enabled"`
+
+    // Minimum period to wait between refreshing remote-write reconfigurations.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:="10s"
+    RefreshPeriod LokiDuration `json:"refreshPeriod"`
+
+    // Defines the configuration for remote write client.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:advanced",displayName="Client"
+    ClientSpec *RemoteWriteClientSpec `json:"clientSpec"`
+
+    // Defines the configuration for remote write client queue.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:advanced",displayName="Client Queue"
+    QueueSpec *RemoteWriteClientQueueSpec `json:"queueSpec"`
+}
+
+// LokiRulerSpec defines the desired state of LokiRuler
+type LokiRulerSpec struct {
```

Comment:
I think this would also be a good place for the selectors that Simon proposed for `LokiRule` instead of using the `StackName`. I like the idea of using matchers to find the rules, so that we are able to re-use rules across multiple stacks.

Reply:
Good point, I've update the proposal by removing this field.